tests/it/compute/cast.rs

```diff
@@ -817,6 +817,43 @@ fn utf8_to_naive_timestamp() {
     assert_eq!(expected, result.as_ref());
 }
 
+#[test]
+fn utf8_to_naive_timestamp_ms() {
+    let array =
+        Utf8Array::<i32>::from_slice(["1996-12-19T16:39:57-02:00", "1996-12-19T17:39:57-02:00"]);
+    // the timezone is disregarded from the string and we assume UTC
+    let expected = Int64Array::from_slice([851013597000, 851017197000])
+        .to(DataType::Timestamp(TimeUnit::Millisecond, None));
+
+    let result = cast(&array, expected.data_type(), CastOptions::default()).expect("cast failed");
+    assert_eq!(expected, result.as_ref());
+}
+
+#[test]
+fn utf8_to_naive_timestamp_us() {
+    let array =
+        Utf8Array::<i32>::from_slice(["1996-12-19T16:39:57-02:00", "1996-12-19T17:39:57-02:00"]);
+    // the timezone is disregarded from the string and we assume UTC
+    let expected = Int64Array::from_slice([851013597000000, 851017197000000])
+        .to(DataType::Timestamp(TimeUnit::Microsecond, None));
+
+    let result = cast(&array, expected.data_type(), CastOptions::default()).expect("cast failed");
+    assert_eq!(expected, result.as_ref());
+}
+
+#[test]
+fn utf8_to_timestamp_ms_with_tz() {
+    let tz = "-02:00".to_string();
+    let array =
+        Utf8Array::<i32>::from_slice(["1996-12-19T16:39:57-02:00", "1996-12-19T17:39:57-02:00"]);
+    // the timezone is used to map the time to UTC.
+    let expected = Int64Array::from_slice([851013597000, 851017197000])
+        .to(DataType::Timestamp(TimeUnit::Millisecond, Some(tz)));
+
+    let result = cast(&array, expected.data_type(), CastOptions::default()).expect("cast failed");
+    assert_eq!(expected, result.as_ref());
```

Comment:
I am getting some issues when trying to implement unit testing for ms and us for timestamp with tz, the expected timestamps get modified by subtracting two hours to the original time, interestingly this does not happen when I use ns time unit, any Idea why ?